src/libData/BlockChainData/BlockChain.h

```diff
@@ -71,13 +71,14 @@ class BlockChain {
   T GetBlock(const uint64_t& blockNum) {
     std::lock_guard<std::mutex> g(m_mutexBlocks);
 
-    if (blockNum >= m_blocks.size()) {
-      LOG_GENERAL(WARNING, "Block number "
-                               << blockNum
-                               << " absent, a dummy block will be used and "
-                                  "abnormal behavior may happen!");
+    if (m_blocks.size() > 0 &&
+        (m_blocks.back().GetHeader().GetBlockNum() < blockNum)) {
```

Comment:
As the vc block and fallback block doesn't have `GetBlockNum()`, maybe we can overidely declare the function `GetBlock` for that two specific block types as deleted or return a dummy value.

Reply:
But in another case, if we deleted that, will it affect the block fetching by using BlockIndexChain?

Reply:
Or we can have a blockNum for vc or fb block incrementally, which I feels it the best solution for now

Reply:
I dont think so this is in the scope of this PR

Reply:
Yeah it's not for add block num. But firstly should avoid vc and fb call GetBlockNum in this PR. otherwise it's undefined behavior

Reply:
VC/FB are not attached for now in the blockchain.h

Reply:
VC/FBBlockChain